packages/cli/src/server.ts

```diff
@@ -429,6 +430,45 @@ export class Server extends AbstractServer {
 				}
 			};
 
+			const extensions = ['@n8n/n8n-extension-cloud'];
+			const loadedExtensions: ExtensionManifest[] = [];
+
+			for (const extension of extensions) {
+				const manifestJSON = await readFile(require.resolve(extension), 'utf-8');
+				const manifest = jsonParse<ExtensionManifest>(manifestJSON);
+
+				loadedExtensions.push(manifest);
+			}
+
+			const serveExtensions: express.RequestHandler = async (req, res) => {
+				res.status(200);
+				res.setHeader('Content-Type', 'application/json');
+				res.send(loadedExtensions);
+			};
+			this.app.use('/rest/extensions', serveExtensions);
+
+			const serveFrontendExtensions: express.RequestHandler = async (req, res) => {
+				const { publisher, name } = req.params;
+				const extension = loadedExtensions.find(
+					(e) => e.publisher === publisher && e.name === name,
+				);
+
+				if (!extension) {
+					res.sendStatus(404);
+					return;
+				}
+
+				const filePath = require.resolve(`@${publisher}/n8n-extension-${name}/frontend`);
+				if (filePath) {
+					try {
+						await fsAccess(filePath);
+						return res.sendFile(filePath, cacheOptions);
+					} catch {}
+				}
+				res.sendStatus(404);
+			};
+			this.app.use('/assets/extensions/:publisher/:name/frontend.js', serveFrontendExtensions);
+
```

Comment:
Let's use controllers and services as in the rest of the endpoints please? Also ideally behind a flag as this is all WIP.